### PR TITLE
Add page numbers categories in motion TOC

### PR DESCRIPTION
**Pull request by tsiegleauq**
Also upgrades pdfmake.

Removed the "Categories" page from motion catalog.
If motions are sorted by identifier or category, a TOC with categories will be exported.

please review and test.
@FinnStutzenstein 
@emanuelschuetze 

Page 1 of the attached example shows a TOC with categories, page 2 shows a TOC without categories.
[toc_example.pdf](https://github.com/OpenSlides/OpenSlides/files/2123866/toc_example.pdf)

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
Travis (jslint) failed...

**Comment by tsiegleauq** [comment]
> How can I get a ToC without categories?

you will (only) get a ToC with categories if you sort your motions by "Identifier" or "Category".
Everything will export a ToC without categories

File: openslides/motions/static/js/motions/site.js [hunk 1]
```diff
@@ -1435,10 +1435,7 @@ angular.module('OpenSlidesApp.motions.site', [
         };
         // Export dialog
         $scope.openExportDialog = function (motions) {
-            ngDialog.open(MotionExportForm.getDialog(motions));
-        };
-        $scope.pdfExport = function (motions) {
```
Comment (FinnStutzenstein) [inline openslides/motions/static/js/motions/site.js]: Please do not delete this function. It is used for a non admins. Those users do not get the dialog but just an export with default settings.

File: openslides/motions/static/js/motions/pdf.js [hunk 2]
```diff
@@ -853,86 +854,113 @@ angular.module('OpenSlidesApp.motions.pdf', ['OpenSlidesApp.core.pdf'])
             };
 
             var createTOContent = function() {
-                var heading = {
+                var toc = [];
+                var exportCategory = tocWithCategory();
```
Comment (FinnStutzenstein) [inline openslides/motions/static/js/motions/pdf.js]: You can just write `var exportCategory = (sorting === 'identifier' || sorting === 'category.prefix')`

**Comment by tsiegleauq** [comment]
@emanuelschuetze @FinnStutzenstein 
should be fine now.

**Comment by emanuelschuetze** [comment]
@tsiegleauq As discussed with @FinnStutzenstein you should rework this pull request by using pageRef feature to make it less complex, see https://github.com/bpampuch/pdfmake/issues/1116

Also please use the discussed format `$categoryPrefix – $categoryName` in TOC.

**Comment by tsiegleauq** [comment]
@emanuelschuetze 

made the required changes. Content, such as styling, is now fully adjustable and should work nicely

**Comment by FinnStutzenstein** [comment]
Hi, I tested again. I get an `Uncaught Unrecognized document structure: {"pageReference":"","style":"tocEntry","alignment":"right"}` error trying to create a PDF with motion without identifiers. You may better use the id instead of the identifiert, because a motion may not have an identifier.

**Comment by tsiegleauq** [comment]
`id` did not exist in a ContentProvider. Created it now.
@FinnStutzenstein 
@emanuelschuetze

**Comment by emanuelschuetze** [comment]
Travis fixed. Changelog updated.
@FinnStutzenstein Please review and merge.

**Comment by FinnStutzenstein** [comment]
Two little things:
- Category names as "prefix - name"
- Show grouping if there is one category and motions without categories

**Comment by tsiegleauq** [comment]
apparently, pdfMake does not like empty tables. Fixed it.

@emanuelschuetze

**Comment by emanuelschuetze** [comment]
@normanjaeckel @FinnStutzenstein You can review later if you like.

**Comment by normanjaeckel** [comment]
Ok.

**Pull request merged**